dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/support/BroadcastClusterInvoker.java

```diff
@@ -47,21 +48,53 @@ public Result doInvoke(final Invocation invocation, List<Invoker<T>> invokers, L
         RpcContext.getContext().setInvokers((List) invokers);
         RpcException exception = null;
         Result result = null;
+        URL url = getUrl();
+        // The value range of broadcast.fail.threshold must be 0～100.
+        // 100 means that an exception will be thrown last, and 0 means that as long as an exception occurs, it will be thrown.
+        int broadcastFailPercent = url.getParameter("broadcast.fail.percent", 100);
+
+        int failThresholdIndex = invokers.size() * broadcastFailPercent / 100;
+        int failIndex = 0;
         for (Invoker<T> invoker : invokers) {
             try {
                 result = invoker.invoke(invocation);
-            } catch (RpcException e) {
-                exception = e;
-                logger.warn(e.getMessage(), e);
+                if(null != result && result.hasException()){
+                    Throwable resultException = result.getException();
+                    if(null != resultException){
+                        exception = getRpcException(result.getException());
+                        if(failIndex == failThresholdIndex){
+                            break;
+                        }else{
+                            logger.warn(exception.getMessage(), exception);

```

Comment:
Would it be more reasonable if add more message like `Exception caught while invoke boardcast cluster` ?